src/backend/access/transam/timeline.c

```diff
@@ -174,11 +175,31 @@ readTimeLineHistory(TimeLineID targetTLI)
 
 		lasttli = tli;
 
+		end = ((uint64) (switchpoint_hi)) << 32 | (uint64) switchpoint_lo;
+
+		/*
+		 * In case of some recovery we can get such history:
+		 * 1 -> 2
+		 *      2 -> 3
+		 * 1 -> 4
+		 *
+		 * Start pointer for timeline 4 is less than start pointer for timeline 3.
+		 * We can remove timeline 3 and get such history: 1 -> 2 -> 4.
```

Comment:
I think that a comment on the code should describe the general approach, rather than show a specific example.

Reply:
Changed.